web/html/src/manager/virtualization/guests/list/guests-list.js

```diff
@@ -0,0 +1,401 @@
+const React = require("react");
+const PropTypes = React.PropTypes;
+const MessageContainer = require("../../../../components/messages").Messages;
+const MessagesUtils = require("../../../../components/messages").Utils;
+const {Table, Column, SearchField, Highlight} = require("../../../../components/table");
+const Functions = require("../../../../utils/functions");
+const Utils = Functions.Utils;
+const {LinkButton, AsyncButton} = require("../../../../components/buttons");
+const {DeleteDialog} = require("../../../../components/dialog/DeleteDialog");
+const {DangerDialog} = require("../../../../components/dialog/DangerDialog");
+const {ModalButton} = require("../../../../components/dialog/ModalButton");
+const Systems = require("../../../../components/systems");
+import VirtualizationGuestActionApi from '../virtualization-guest-action-api';
+import VirtualizationGuestsListRefreshApi from '../virtualization-guests-list-refresh-api';
+
+class GuestsList extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      actionsResults: {},
+      errors: [],
+      selectedItems: []
+    }
+
+    this.onBeforeUnload = (event) => {
+        this.setState({
+            pageUnloading: true
+        });
+    };
+
+    this.handleSelectItems = (items) => {
+      this.setState({
+        selectedItems: items
+      });
+    };
+
+    this.selectGuest = (row) => {
+      this.setState({
+        selected: row
+      });
+    };
+
+    this.actionCallback = (results) => {
+      var actions = this.state.actionsResults;
+      Object.entries(results).forEach(([uuid, actionid]) => {
+        actions[uuid] = {'id': actionid, 'result': 'Queued'};
+      });
+      this.setState({actionsResults: actions});
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.salt_entitled) {
+      var port = window.location.port;
+      var url = "wss://" +
+        window.location.hostname +
+        (port ? ":" + port : "") +
+         "/rhn/websocket/minion/virt-notifications";
+      var ws = new WebSocket(url);
+
+      ws.onopen = () => {
```

Comment:
Point for future:

is it expected that the logic of this page will grow? IMO more than 400 lines in one file will start to become unmanageable. If it does in the future it would make sense to split the websocket logic outside in the same way we did for the api calls

Reply:
no, that page is already feature complete, shouldn't grow